git_machete/tests/functional/test_machete.py

```diff
@@ -72,74 +104,60 @@ def handle_get(self) -> None:
                 return
             else:
                 self.status_code = 200
-                self.return_data = GitAPIState.pulls
+                self.return_data = MockGitAPIRequest.git_api_state.pulls
                 return
-        elif 'user' in self.url:
+        elif self.parsed_url.path.endswith('user'):
             self.status_code = 200
-            self.return_data = GitAPIState.user
+            self.return_data = MockGitAPIRequest.git_api_state.user
             return
 
     def handle_patch(self) -> None:
-        if 'issues' in self.url:
+        if 'issues' in self.parsed_url.path:
             self.update_issue()
-        elif 'pulls' in self.url:
+        elif 'pulls' in self.parsed_url.path:
             self.update_pull()
 
     def handle_post(self) -> None:
-        if 'issues' in self.url:
-            if self.url.endswith('issues'):
+        if self.parsed_url.path.endswith('issues'):
+            if not self.parsed_query:
```

Comment:
In fact since you're NOT expecting any query params to be present (I guess that's the intention here), you can add an assertion